pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -176,7 +174,7 @@ rec {
         // {
           # A hack to make `nix-env -qa` and `nix search` ignore broken packages.
           # TODO(@oxij): remove this assert when something like NixOS/nix#1771 gets merged into nix.
-          name = assert validity.handled; name + lib.optionalString
+          name = assert (validity.handled && separateDebugInfo -> stdenv.hostPlatform.isLinux); name + lib.optionalString
```

Comment:
I think we can just leave this off. It's awkward to put it here.

Reply:
But then I'd be removing the assertion altogether? if `check-meta` checked things besides `meta` that would clean this up, but that's a larger refactor.

Reply:
It just doesn't seem like a reason for a package to be "broken". Maybe replace ```separateDebugInfo``` references above with ```separateDebugInfo && stdenv.hostPlatform.isLinux```?

Reply:
But then I'm still changing the interface? Also asking for `separateDebugInfo` and not getting it could be confusing, and finally re-tightening the interface after its so loosened is a breaking change.

Reply:
2a28bc669124190e28f7d62663eee9ce31759bc8 is the origin of the assert, BTW.